themes/default/content/registry/packages/azuread/installation-configuration.md

```diff
@@ -119,20 +99,71 @@ To list all available subscriptions, you can use:
+    $ pulumi config set azuread:tenantId <tenantID>
+    $ pulumi config set azuread:subscriptionId <subscriptionId>
+    ```
+
+##### Set configuration using environment variables
+
+{{< chooser os "linux,macos,windows" >}}
+{{% choosable os linux %}}
 
 ```bash
-$ export ARM_CLIENT_ID="WWWWWWWW-WWWW-WWWW-WWWW-WWWWWWWWWWWW"
-$ export ARM_CLIENT_SECRET="XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX"
-$ export ARM_TENANT_ID="YYYYYYYY-YYYY-YYYY-YYYY-YYYYYYYYYYYY"
+$ export ARM_CLIENT_ID=<YOUR_ARM_CLIENT_ID>
+$ export ARM_CLIENT_SECRET=<YOUR_ARM_CLIENT_SECRET>
+$ export ARM_TENANT_ID=<YOUR_ARM_TENANT_ID>
+$ export ARM_SUBSCRIPTION_ID=<YOUR_ARM_SUBSCRIPTION_ID>
 ```
 
-Or configuration variables, if you prefer that they be stored alongside your Pulumi stack for easy multi-user access:
+{{% /choosable %}}
+
+{{% choosable os macos %}}
 
 ```bash
-$ pulumi config set azuread:clientId "WWWWWWWW-WWWW-WWWW-WWWW-WWWWWWWWWWWW"
-$ pulumi config set azuread:clientSecret "XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX" --secret
-$ pulumi config set azuread:tenantId "YYYYYYYY-YYYY-YYYY-YYYY-YYYYYYYYYYYY"
+$ export ARM_CLIENT_ID=<YOUR_ARM_CLIENT_ID>
+$ export ARM_CLIENT_SECRET=<YOUR_ARM_CLIENT_SECRET>
+$ export ARM_TENANT_ID=<YOUR_ARM_TENANT_ID>
+$ export ARM_SUBSCRIPTION_ID=<YOUR_ARM_SUBSCRIPTION_ID>
+```
+
+{{% /choosable %}}
+
+{{% choosable os windows %}}
+
+```powershell
+> $env:ARM_CLIENT_ID = "<YOUR_ARM_CLIENT_ID>"
+> $env:ARM_CLIENT_SECRET = "<YOUR_ARM_CLIENT_SECRET>"
+> $env:ARM_TENANT_ID = "<YOUR_ARM_TENANT_ID>"
+> $env:ARM_SUBSCRIPTION_ID = "<YOUR_ARM_SUBSCRIPTION_ID>"
 ```
 
-Remember to pass `--secret` when setting `clientSecret` so that it is properly encrypted.
+{{% /choosable %}}
+{{< /chooser >}}
+
+## Configuration options
+
+Use `pulumi config set azuread:<option>` or pass options to the [constructor of `new azuread.Provider`]({{< relref "/registry/packages/azure/api-docs/provider" >}}).
+
+| Option                      | Required/Optional | Description                                                                                                                                                                                                                                                                                            |
+|-----------------------------|-------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `environment`               | Required          | The cloud environment to use. It can also be sourced from the ARM_ENVIRONMENT environment variable. Supported values are: `public` (default), `usgovernment`, `china`.                                                                                                                                 |
```

Comment:
Is it really required when there's a default?

Reply:
I have found this particular reference column to be very confusing myself! 
I came down on leaving it as Required - if you have an environment set, implicitly or otherwise, that doesn't really change that `environment` is a necessary configuration for Pulumi.

Reply:
You could add "Defaults to `public`."

Reply:
It already says that in the description - do you mean adding it very obviously in the Required/Optional column?

Reply:
I overlooked that, nvm